design/ingressroute-design.md

```diff
@@ -387,9 +387,31 @@ The best way I've thought to mitigate this so far is the vertex ingress routes w
 
 This is unfortunate for two reasons:
 
-- It's more boilerplate: a deletates to b, b has to list a as a cross check.
+- It's more boilerplate: a delegates to b, b has to list a as a cross check.
 - It turns a DAG into a linked list (possibly not the right term), but it would be impossible for a web service to be a member of multiple roots, unless of course we made the list of incoming vertices be a list -- which would probably push the solution into using virtualhost.fqdn.
 
 The last thing is that this boilerplate _should be required_ even when not in enforcing mode.
 I'm not interested in proposing a design where the security interlock that prevents dave's cheap webhosting from publishing gmail without TLS is considered to be the customer's problem because they did not add an optional key. 
 Said again, if this is the mitigation we choose to adopt, it has to be mandatory for all users, because we all know how effective optional security features are.
+
+# Metrics
+
+Metrics are essential to any system. Contour will expose a `/metrics` Prometheus endpoint with the following metrics:
+
+- **contour_ingressroute_total (gauge):** Number of IngressRoutes
+  - namespace
+  - vhost
+- **contour_ingressroute_upstream_total (gauge):** Number of Upstreams per IngressRoute
+  - namespace
```

Comment:
Is this is the namespace and name of the IngressRoute resource? Just want to make sure I am following.

Reply:
Correct, `namepace==where the IngressRoute is deployed` and `name==the name of the IngressRoute`. The only other exception is the ingressroute_total which is tagged by `vhost`.

Reply:
Got it. Thanks!